[DBZ-1715] Add route by field option for ExtractRecordState transformation

jhuiting opened the pull request:
https://issues.redhat.com/browse/DBZ-1715

As explained in DBZ-1715 this SMT adds the ability to set a topic based on the field in the table this SMT is consuming. 

I've checked on the Gitter dev channel if this is potentially interesting for core and @Naros replied that it might be. I've added some basic tests, would like to know how much more testing coverage is required for this piece of code.

Naros commented:
@jhuiting Can you take a look at `ExtractNewRecordState`, there appears to be a formatting problem.  If you want the build to format the file for you, you can run the following command locally and then check the changed files after and commit them if they look ok:

```
mvn clean install -pl debezium-core -am -Passembly
```

jhuiting commented:
@Naros Pushed a fix, should be all good. Thanks for the command, it complained before that the formatter was missing.

gunnarmorling commented on debezium-core/src/test/java/io/debezium/transforms/ExtractNewRecordStateTest.java:
```diff
@@ -294,6 +304,34 @@ public void testAddSourceField() {
     }
 
+    @Test
+    public void testAddTopicRoutingField() {
```
Could you add a test to make sure that the new column value is applied in case of an update event?

jhuiting replied:
Done

jhuiting commented:
@gunnarmorling Added the testcase you requested, squashed the commits and added the config option to the docs. Should be ready for another review round.

gunnarmorling commented:
Excellent, thank you so much!
